src/backend/utils/mmgr/memprot.c

```diff
@@ -245,20 +265,9 @@ GPMemoryProtect_TrackStartupMemory(void)
 	 */
 	bytes += 6L << BITS_IN_MB;
 
-#ifdef USE_ORCA
-	/* When compile with ORCA it will commit 6MB more */
-	bytes += 6L << BITS_IN_MB;
-
-	/*
-	 * When optimizer_use_gpdb_allocators is on, at least 2MB of above will be
-	 * tracked by vmem tracker later, so do not recount them.  This GUC is not
-	 * available until gpdb 5.1.0 .
-	 */
-#if GP_VERSION_NUM >= 50100
-	if (optimizer_use_gpdb_allocators)
-		bytes -= 2L << BITS_IN_MB;
-#endif  /* GP_VERSION_NUM */
-#endif  /* USE_ORCA */
+	/* freeze the addin request size and include it */
+	startup_mem_addin_request_allowed = false;
+	bytes = add_size(bytes, startup_mem_total_addin_request);
```

Comment:
I'm not sure if it's right to use the `Size` type (aka size_t) for `startup_mem_total_addin_request` since `bytes` are `int64`. I think it makes sense to use the same types (or add an explicit type cast).

Reply:
I've changed `bytes` in the GPMemoryProtect_TrackStartupMemory() to Size, and added type cast when passing it to the `VmemTracker_RegisterStartupMemory()`.

I think that using signed `int64` for memory size is a quite controversial decision... But as not to rewrite the entire code in the vmem_tracker, I've limited the delta with the changes I've described.

Reply:
Should we then use `add_size` instead of simply summing in the entire function in such a case? At least in the next one. Or make adding `startup_mem_total_addin_request` the last one.

Reply:
let's make the `startup_mem_total_addin_request` the last one.